src/ripple/core/SQLInterface_global.h

```diff
@@ -0,0 +1,204 @@
+
+void
+loadManifest(
+    soci::session& dbCon,
+    std::string const& dbTable,
+    beast::Journal& j,
+    ManifestCache& mCache);
+
+void
+saveManifest(
+    soci::session& dbCon,
+    std::string const& dbTable,
+    std::function<bool(PublicKey const&)> isTrusted,
+    beast::Journal& j,
+    hash_map<PublicKey, Manifest>& map);
+
+void
+addValidatorManifest(soci::session& db, std::string const& serialized);
+
+std::pair<PublicKey, SecretKey>
+loadNodeIdentity(soci::session& db);
+
+void
+loadPeerReservationTable(
+    soci::session& conn,
+    beast::Journal& j,
+    std::unordered_set<PeerReservation, beast::uhash<>, KeyEqual>& table);
+
+void
+insertPeerReservation(
+    soci::session& conn,
+    PublicKey const& nodeId,
+    std::string const& description);
+
+void
+deletePeerReservation(soci::session& conn, PublicKey const& nodeId);
+
+/* Shard asquire DB */
+
+std::unique_ptr<DatabaseCon>
+makeAcquireDB(
+    Config const& config,
+    boost::filesystem::path const& dir,
+    DatabaseCon::CheckpointerSetup const& checkpointerSetup);
+
+void
+insertAcquireDBIndex(soci::session& db, std::uint32_t index_);
+
+std::pair<bool, boost::optional<std::string>>
+selectAcquireDBLedgerSeqs(soci::session& db, std::uint32_t index);
+
+std::tuple<bool, boost::optional<std::string>, boost::optional<std::string>>
+selectAcquireDBLedgerSeqsHash(soci::session& db, std::uint32_t index);
+
+void
+updateAcquireDB(
+    soci::session& db,
+    std::shared_ptr<Ledger const> const& ledger,
+    std::uint32_t const index,
+    std::uint32_t const lastSeq,
+    boost::optional<std::string> seqs);
+
+/* Archive DB */
+
+std::unique_ptr<DatabaseCon>
+makeArchiveDB(boost::filesystem::path const& dir, std::string const& dbName);
+
+void
+readArchiveDB(
+    soci::session& db,
+    std::function<void(std::string const&, int)> const& func);
+
+void
+insertArchiveDB(
+    soci::session& db,
+    LedgerIndex shardIndex,
+    std::string const& url);
+
+void
+deleteFromArchiveDB(soci::session& db, LedgerIndex shardIndex);
+
+void
+dropArchiveDB(soci::session& db);
+
+/* State DB */
+
+struct SavedState
+{
+    std::string writableDb;
+    std::string archiveDb;
+    LedgerIndex lastRotated;
+};
+
+void
+initStateDB(
+    soci::session& db,
+    BasicConfig const& config,
+    std::string const& dbName);
+
+LedgerIndex
+getCanDelete(soci::session& db);
+
+LedgerIndex
+setCanDelete(soci::session& db, LedgerIndex canDelete);
+
+SavedState
+getSavedState(soci::session& db);
+
+void
+setSavedState(soci::session& db, SavedState const& state);
+
+void
+setLastRotated(soci::session& db, LedgerIndex seq);
+
+/* DatabaseBody DB */
+
+std::pair<std::unique_ptr<DatabaseCon>, boost::optional<std::uint64_t>>
+openDatabaseBodyDb(Config const& config, boost::filesystem::path const& path);
+
+void
+databaseBodyDoPut(
+    soci::session& conn,
+    std::string& data,
+    std::string& path,
+    std::uint64_t fileSize,
+    std::uint64_t& part,
+    std::uint16_t const maxRowSizePad);
```

Comment:
In general, we don't pass value types by `const` (for example, `fileSize` in this function is not `const`). I do understand the motivations to do so, but I'd prefer to be consistent and don't use `const` on value type.

Reply:
Done.